File: ee355/hw2/hw2.cpp
#include <iostream>
#include <stdio.h>


int main() {
   int money, orphan;
   int z = 0;
   std::cout << "How much money do you wanna donate? " ;
   std::cin >> money;
   std::cout << "how many orphans will you be donating this money to? " ;
   std::cin >> orphan;
   
   int array [orphan];

   while (money!=0) 
   {
      for (int i = 1; i < (orphan+1); i++) 
      {
         if (z == 0) 
         {
            if (money >= i)
            {
               money = money - i;
               array[i-1] = i;              
            }
            else 
            {
               if (money!=0) 
               {
                  array[i-1] = money;
                  money=0;
               }
               else
                  array[i-1] = 0;
            }
         }
         else 
         {
            if (money >= (i+z)) 
            {
               money = money - (i+z);
               array[i-1] = i + z + array[i-1];
            }
            else 
            {
               if (money!=0) 
               {
                  array[i-1] = money;
                  money = 0;
                  break; 
               }
               else
                  break;
            }
         }
         }
         z++;
      }
   int i=0;
   for (int k = 1; k < (orphan+1) ; k++) 
   {
      std::cout << "Orphan #" << k << " gets " << array[i] << " dollars\n";
      i++;
   }
   return 0; 
}      
            
          
          
               
            

File: ee355/hw2/hw2.py
x = input("How much money do you wanna donate?")
y = input("How many orphans will you be donating this money to?")

k = []
z = 0


while x != 0:
   for i in range(1, y+1):
      if z == 0:
         if x >= i:
            x = x - i
            ins = i 
            k.append(ins)
         else:
            if x != 0:
               ins = x 
               x=0
               k.append(ins)
            else:
               k.append(0)
      else:
         if x >= (i + z):
            x = x -(i+z)
            k[i-1] = i + z + k[i-1]
         else:
            if x != 0:
               k[i-1] = x + k[i-1]
               x=0
               break
            else:
               break
   z += 1
for i in range(1,y+1):
   print("Orphan #%i gets %i dollars" % (i,k[i-1]))
